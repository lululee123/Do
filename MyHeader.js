import React from 'react';
import { StyleSheet, Text, View, Image, TouchableWithoutFeedback } from 'react-native';

const MyHeader = props => {
  return (
    <View style={styles.titleBar}>
      {
        props.menu ? 
        <TouchableWithoutFeedback  onPress={() => props.navigation.toggleDrawer()} >
          <Image style={styles.imageStyle} source={require('./image/menu.png')} />
        </TouchableWithoutFeedback> :
        <View></View>
      }
      <Text style={styles.titleBarText}>{props.title}</Text>
    </View>
  );
};

export default MyHeader;

const styles = StyleSheet.create({
  titleBar: {
    height: 80,
    backgroundColor: '#3A3D5E',
    paddingTop: 30,
    flexDirection: 'row',
    alignItems: 'center',
    textAlign: 'center',
    position: 'relative',
  },
  titleBarText: {
    textAlign: 'center',
    color: '#fff',
    fontSize: 24,
    fontWeight: "bold",
    flex: 1
  },
  imageStyle: {
    width: 30, 
    height: 22, 
    position: 'absolute',
    top: 45,
    left: 20, 
    zIndex: 1
  }
});