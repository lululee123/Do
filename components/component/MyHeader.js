import React from 'react';
import { StyleSheet, Text, View, Image, TouchableWithoutFeedback, Button } from 'react-native';

const MyHeader = props => {
  return (
    <View style={styles.titleBar}>
      {
        props.menu ? 
        <TouchableWithoutFeedback  onPress={() => props.navigation.toggleDrawer()} >
          <Image style={styles.imageStyle} source={require('../../image/menu.png')} />
        </TouchableWithoutFeedback> :
        <View></View>
      }
      <Text style={styles.titleBarText}>{props.title}</Text>
      {
        props.search ? 
        <View style={styles.searchBtn}>
          <Button
            color='white'
            title="&#128269;"
            onPress={() => props.navigation.navigate('Search')}
          />
        </View> : 
        <View></View>
      }
    </View>
  );
};

export default MyHeader;

const styles = StyleSheet.create({
  titleBar: {
    height: 80,
    backgroundColor: '#3A3D5E',
    paddingTop: 30,
    flexDirection: 'row',
    alignItems: 'center',
    textAlign: 'center',
    position: 'relative',
  },
  titleBarText: {
    textAlign: 'center',
    color: '#fff',
    fontSize: 24,
    fontWeight: "bold",
    flex: 1
  },
  imageStyle: {
    width: 30, 
    height: 22, 
    position: 'absolute',
    top: 45,
    left: 20, 
    zIndex: 1
  },
  searchBtn: {
    position: 'absolute',
    width: 30, 
    height: 30, 
    right: 20,
    top: 35,
    zIndex: 1
  }
});