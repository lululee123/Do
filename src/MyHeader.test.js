import React from 'react';
import { shallow, configure } from 'enzyme';
import MyHeader from '../components/component/MyHeader';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() })

describe('Rendering', () => {
    it('render without any props', () => {
        const component = shallow(<MyHeader />);
        expect(component).toMatchSnapshot();
    });

    it('title display correct', () => {
        const titleArray = ['', '123', 'abc'];
        titleArray.map( item => {
            const component = shallow(<MyHeader title={item} />);
            expect(component.find('Text').at(0).props().children).toBe(item);   
            expect(component).toMatchSnapshot();  
        })
    })

    it('menu display', () => {
        const component = shallow(<MyHeader menu={true} />);
        expect(component.find('TouchableWithoutFeedback').length).toBe(1);
        expect(component.find('Image').length).toBe(1);
        expect(component).toMatchSnapshot();  
    })

    it('menu hidden', () => {
        const component = shallow(<MyHeader menu={false} />);
        expect(component.find('View').length).toBe(3);
        expect(component).toMatchSnapshot();  
    })

    it('search display', () =>{
        const component = shallow(<MyHeader search={true} />);
        expect(component.find('Button').length).toBe(1);
        expect(component).toMatchSnapshot();
    })

    it('search hidden', () =>{
        const component = shallow(<MyHeader search={false} />);
        expect(component.find('Button').length).toBe(0);
        expect(component).toMatchSnapshot();
    })

    it('menu and search both display', () => {
        const component = shallow(<MyHeader menu={true} search={true} />);
        expect(component.find('TouchableWithoutFeedback').length).toBe(1);
        expect(component.find('Image').length).toBe(1);
        expect(component.find('View').length).toBe(2);
        expect(component.find('Button').length).toBe(1);
        expect(component).toMatchSnapshot();
    })

    it ('menu press', () => {
        const goToFunc = jest.fn();
        const navigation = { toggleDrawer: goToFunc };
        const component = shallow(<MyHeader menu={true} navigation={ navigation } />);
        component.find('TouchableWithoutFeedback').at(0).simulate('press');
        expect(goToFunc).toHaveBeenCalled(); 
    })

    it('search press', () => {
        const goToFunc = jest.fn();
        const navigation = { navigate: goToFunc };
        const component = shallow(<MyHeader search={true} navigation={ navigation } />);
        component.find('Button').at(0).simulate('press');
        expect(goToFunc).toHaveBeenCalled(); 
        expect(goToFunc).toHaveBeenCalledWith('Search');   
    })
});

