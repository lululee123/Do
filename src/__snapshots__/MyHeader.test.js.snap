// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering menu and search both display 1`] = `ShallowWrapper {}`;

exports[`Rendering menu display 1`] = `ShallowWrapper {}`;

exports[`Rendering menu hidden 1`] = `ShallowWrapper {}`;

exports[`Rendering render without any props 1`] = `ShallowWrapper {}`;

exports[`Rendering search display 1`] = `ShallowWrapper {}`;

exports[`Rendering search hidden 1`] = `ShallowWrapper {}`;

exports[`Rendering title display correct 1`] = `ShallowWrapper {}`;

exports[`Rendering title display correct 2`] = `ShallowWrapper {}`;

exports[`Rendering title display correct 3`] = `ShallowWrapper {}`;
